src/main/java/com/arpnetworking/metrics/mad/model/statistics/HistogramStatistic.java

```diff
@@ -375,9 +375,10 @@ public Double getValueAtPercentile(final double percentile) {
             // The Math.min is for the case where the computation may be just
             // slightly larger than the _entriesCount and prevents an index out of range.
             final int target = (int) Math.min(Math.ceil(_entriesCount * percentile / 100.0D), _entriesCount);
-            int accumulated = 0;
-            for (final Double2IntMap.Entry next : _data.double2IntEntrySet()) {
-                accumulated += next.getIntValue();
+            // TODO(ville): Is it sufficient for this to be a long?
```

Comment:
`accumulated` will be bounded by the count of the histogram, right? So if the count can be a `long`, this can too?

Reply:
Yes, `accumulated` here and `count` in the augmented-histogram both represent the total number of samples represented. Both should be `long`.  

My question here is more forward looking that if individual bucket counts can overflow an `int` is `accumulated` (and `count`) sufficiently large as a `long`? Will we ever have histograms with more than 9,223,372,036,854,775,808 samples? 

=> I think not, it would be 42M samples per minute per client into a daily roll-up over 150M clients.